Project-Hotpot/Project-Hotpot/Algorithm/SongAlgorithm.swift

```diff
@@ -14,7 +14,33 @@ class SongAlgorithm{
     var userPreferences = UserSettingsManager.shared()
     var genreQueueManager = GenreQueueManager.shared()
     var cacheManager = CacheManager.shared()
+    var scoreManager = SongScoreManager.shared()
+    var songManager = SongManager()
     
+    
+    func playNewSong(completion: @escaping (_ result: Result<Void, Error>) -> Void) {
```

Comment:
Hmm probably does not belong here. A SongAlgorithm should provide what to play, some other manager or controller should ask the algorithm and invoke play. I will let you figure out whats the best place